pkg/ucp/frontend/controller/credentials/azure/createorupdateazurecredential.go

```diff
@@ -79,15 +80,30 @@ func (c *CreateOrUpdateAzureCredential) Run(ctx context.Context, w http.Response
 		newResource.Properties.Storage.InternalCredential.SecretName = secretName
 	}
 
-	// Save the credential secret
-	err = secret.SaveSecret(ctx, c.secretClient, secretName, newResource.Properties.AzureCredential)
-	if err != nil {
-		return nil, err
+	switch newResource.Properties.Kind {
+	case datamodel.AzureServicePrincipalCredentialKind:
+		if newResource.Properties.AzureCredential.ServicePrincipal == nil {
+			return armrpc_rest.NewBadRequestResponse("Invalid Service Principal Credential"), nil
+		}
+		// Save the credential secret
+		err = secret.SaveSecret(ctx, c.secretClient, secretName, newResource.Properties.AzureCredential)
+		if err != nil {
+			return nil, err
+		}
+		newResource.Properties.AzureCredential.ServicePrincipal.ClientSecret = ""
+	case datamodel.AzureWorkloadIdentityCredentialKind:
+		if newResource.Properties.AzureCredential.WorkloadIdentity == nil {
+			return armrpc_rest.NewBadRequestResponse("Invalid Workload Identity Credential"), nil
+		}
+		// Save the credential secret
+		err = secret.SaveSecret(ctx, c.secretClient, secretName, newResource.Properties.AzureCredential)
+		if err != nil {
+			return nil, err
+		}
+	default:
+		return armrpc_rest.NewBadRequestResponse("Invalid Credential Kind"), nil
 	}
 
-	// Do not save the secret in metadata store.
-	newResource.Properties.AzureCredential.ClientSecret = ""
```

Comment:
Do we need to keep this code that blanks out the client secret? I think this was here to prevent the client secret from getting saved to the DB in the service principal case.

Reply:
I think so, I just moved it into the service principal switch because workload identity doesn't need it: https://github.com/radius-project/radius/blob/ff78c0ac5154a5b46063341bc1513ff9e902829c/pkg/ucp/frontend/controller/credentials/azure/createorupdateazurecredential.go#L93